tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -64,7 +67,13 @@ std::unique_ptr<ROOT::Experimental::Detail::RPageSource> ROOT::Experimental::Det
 #else
       throw RException(R__FAIL("This RNTuple build does not support DAOS."));
 #endif
-
+   if (location.find("s3://") == 0) {
+#ifdef R__ENABLE_DAVIX
+      return std::make_unique<RPageSourceS3>(ntupleName, location, options);
+#else
+      throw RException(R__FAIL("This RNTuple build does not support S3."));
+#endif
+   }
```

Comment:
TODO: if we are going to support many backends in the future, maybe we should make the (conditional) construction of the `RPageStorageXxx` a bit more elegant.